controllers/clustermesh/clustermesh_controller.go

```diff
@@ -123,63 +195,125 @@ func (r *ClusterMeshReconciler) reconcileDelete(ctx context.Context, cluster *cl
-	clSpec, err := r.PopulateClusterSpecFactory(r, ctx, cluster)
+func ReconcileSecurityGroups(r *ClusterMeshReconciler, ctx context.Context, clustermesh *clustermeshv1beta1.ClusterMesh) error {
+	// Update status
+	ownedSecurityGroupsRefs, err := crossplane.GetOwnedSecurityGroups(ctx, clustermesh, r.Client)
 	if err != nil {
-		return ctrl.Result{}, err
+		return err
 	}
+	clustermesh.Status.CrossplaneSecurityGroupRef = ownedSecurityGroupsRefs
 
-	key := client.ObjectKey{
-		Name: cluster.Labels["clusterGroup"],
+	sgPrefix := "clustermesh-"
+	cidrResults := map[string][]string{}
+	for _, cl := range clustermesh.Spec.Clusters {
+
+		// Get control plane ASG
+		kcp := &kcontrolplanev1alpha1.KopsControlPlane{}
+		key := client.ObjectKey{
+			Namespace: "kubernetes-" + cl.Name,
+			Name:      cl.Name,
+		}
+		if err := r.Client.Get(ctx, key, kcp); err != nil {
+			return err
+		}
+		cidrResults[cl.Name] = append(cidrResults[cl.Name], kcp.Spec.KopsClusterSpec.NetworkCIDR)
 	}
 
-	if err := r.Get(ctx, key, clustermesh); err != nil {
-		if !errors.IsNotFound(err) {
-			return ctrl.Result{}, err
+	for _, cl := range clustermesh.Spec.Clusters {
+		key := client.ObjectKey{
+			Name: sgPrefix + cl.Name,
 		}
+		kopsControlPlaneSG := &sgv1beta1.SecurityGroup{}
+		if err := r.Get(ctx, key, kopsControlPlaneSG); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
 
-		ccm := crossplane.NewCrossPlaneClusterMesh(cluster.Labels["clusterGroup"], clSpec)
-		r.log.Info(fmt.Sprintf("creating clustermesh %s", ccm.ObjectMeta.GetName()))
-		if err := r.Create(ctx, ccm); err != nil {
-			return ctrl.Result{}, err
+			kopsControlPlaneSG.Name = sgPrefix + cl.Name
+			kopsControlPlaneSG.Namespace = "kubernetes-" + cl.Name
+
+			// Get nodes asg
+			kopsControlPlaneSG.Spec.InfrastructureRef = &corev1.ObjectReference{
+				APIVersion: "controlplane.cluster.x-k8s.io/v1alpha1",
+				Kind:       "KopsControlPlane",
+				Name:       kopsControlPlaneSG.Name,
+				Namespace:  kopsControlPlaneSG.Namespace,
+			}
+
+			kopsControlPlaneSG.Spec.IngressRules = []sgv1beta1.IngressRule{
+				{
+					IPProtocol:        "udp",
+					FromPort:          0,
+					ToPort:            65000,
+					AllowedCIDRBlocks: cidrResults[cl.Name]},
+			}
+			r.log.Info(fmt.Sprintf("creating security group %s for control plane", kopsControlPlaneSG.ObjectMeta.GetName()))
+			if err := r.Create(ctx, kopsControlPlaneSG); err != nil {
+				return err
+			}
 		}
-		return ctrl.Result{}, nil
-	} else {
+	}
 
-		if !r.isClusterBelongToMesh(cluster.Name, *clustermesh) {
-			clustermesh.Spec.Clusters = append(clustermesh.Spec.Clusters, clSpec)
-			r.log.Info(fmt.Sprintf("adding %s to clustermesh %s", cluster.ObjectMeta.Name, clustermesh.Name))
-			// TODO: Verify if we need this with Patch
-			if err := r.Client.Update(ctx, clustermesh); err != nil {
-				return ctrl.Result{}, err
+	for _, cl := range clustermesh.Spec.Clusters {
```

Comment:
This part is the same as the above, shouldn't  this part of the reconciliation contemplate the update of the existing sgs?